lib/charms/layer/kubernetes_common.py

```diff
@@ -1169,7 +1169,7 @@ def configure_kubelet(
     if kube_version >= (1, 19):
         # NB: required for CIS compliance
         feature_gates["RotateKubeletServerCertificate"] = True
-    if is_state("kubernetes-worker.gpu.enabled"):
+    if is_state("kubernetes-worker.gpu.enabled") and kube_version <= (1, 27, 0):
```

Comment:
So?  what about 1.27.1 and beyond?  Seems we'd still want to have this enabled right?

Looking at the [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)

it does seem that starting in 1.28, this feature gate is removed.  Maybe we should do this?

```suggestion
    if is_state("kubernetes-worker.gpu.enabled") and kube_version <= (1, 28, 0):
```

Reply:
Looking at [Feature Gates (removed)](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/) it could also be argued that it could just be < 1.27.0 I believe.